### PR TITLE
fix(language-service): stylable completions should be sorted above css completions

### PR DESCRIPTION
**Pull request by tomrav**
### Tasks

- [x] - set Stylable completions to a higher priority compared to the native CSS ones
- [x] - dedupe resulting completions giving Stylable priority
- [x] - create in-memroy testing mechanism for simpler testing
- [x] - add e2e test in `stylable-intelligence` - after merge and publish test can be merged downstream (wix/stylable-intelligence#756)

### CODE REVIEW

File: packages/language-service/src/lib/service.ts [hunk 1]
```diff
@@ -295,41 +294,59 @@ export class StylableLanguageService {
...
-                                  Math.max(position.character - 1, 0)
-                              ),
-                              position
-                          ),
-                    typeof com.insertText === 'string' ? com.insertText : com.insertText.source
+        const groupedCompletions = new Map<string, CompletionItem>();
+
+        for (const stComp of stCompletions) {
+            const lspCompletion = CompletionItem.create(stComp.label);
+            const ted: TextEdit = TextEdit.replace(
```
Comment (tzachbon) [inline packages/language-service/src/lib/service.ts:301]: What does `ted` means?

File: packages/language-service/src/lib/service.ts [hunk 2]
```diff
@@ -295,41 +294,59 @@ export class StylableLanguageService {
...
-                return lspCompletion;
-            })
-            .concat(this.cssService.getCompletions(cleanDocument, position));
+            } else if (stComp.triggerSignature) {
+                lspCompletion.command = Command.create(
+                    'additional',
+                    'editor.action.triggerParameterHints'
+                );
+            }
+            if (!groupedCompletions.has(lspCompletion.label)) {
```
Comment (tzachbon) [inline packages/language-service/src/lib/service.ts:331]: I guess this `if` could be moved to line 301.

File: packages/language-service/src/lib/service.ts (hunk 1 above)
Comment (tomrav) [reply packages/language-service/src/lib/service.ts:301]: "text edit"

File: packages/language-service/src/lib/service.ts (hunk 2 above)
Comment (tomrav) [reply packages/language-service/src/lib/service.ts]: moved up

File: packages/language-service/src/lib/service.ts [hunk 3]
```diff
@@ -295,41 +294,63 @@ export class StylableLanguageService {
...
+
+        const cssCompletions = this.cssService.getCompletions(cleanDocument, position);
+
+        for (const cssComp of cssCompletions) {
+            const label = cssComp.label;
+
+            if (!groupedCompletions.has(label)) {
+                // CSS declaration property names have built in sorting
+                // at-rules, rules and declaration values do not
+                cssComp.sortText = cssComp.sortText || 'z';
```
Comment (barak007) [inline packages/language-service/src/lib/service.ts:348]: Does this prevent compilation to be sorted by the label?

Comment (tomrav) [reply packages/language-service/src/lib/service.ts:348]: No, we still beat it in the sort - this maintains their internal sorting.

**Review by barak007 — APPROVED** [review]

**Pull request merged**